### ISSUE TITLE
Loss is nan, stopping training

### ISSUE THREAD

**Issue by tau-yihouxiang** [LTH14/mar#69]
When I fine-tune MAR-B as a video generator, there is always NaN loss, and stop training:
```
[11:41:33.534863] Epoch: [8]  [ 200/2293]  eta: 1:52:52  lr: 0.000004  loss: 0.2606 (0.2654)  time: 3.0808  data: 0.0002  max mem: 32637
[11:41:59.064379] Loss is nan, stopping training
E1030 11:42:16.325000 140014536193856 torch/distributed/elastic/multiprocessing/api.py:833] failed (exitcode: 1) local_rank: 0 (pid: 155690) of binary: /data/taohu/anaconda3/envs/vae/bin/python3.10
```
I tried to change the hyper-parameters and nan loss is unavoidable:
--blr 1.0e-5 / --grad_clip 1.0 / larger batch size

What are the possible reasons?

**Comment by LTH14** [comment]
Try to disable the autocast?

**Comment by tau-yihouxiang** [comment]
Thank you for your advice! After disabling autocast, the NaN loss issue has been resolved. However, I've noticed that the training process has become around 80% slower. I am trying to find out the possible reasons.

**Comment by LTH14** [comment]
You could also consider using bf16, which has better numerical stability than fp16

**Comment by tau-yihouxiang** [comment]
Solved! Thank you for the suggestion.

**Comment by drx-code** [comment]
Hi, I also encounter NaN loss and I tried using bf16 training, However, the loss increased sharply, did you met this problem?
<img width="1307" alt="image" src="https://github.com/user-attachments/assets/0bd90467-2ce5-48bd-9d08-d3ade68b259e">

**Comment by tau-yihouxiang** [comment]
Maybe you need to train from scratch.
